overview.tt

```diff
@@ -0,0 +1,279 @@
+          else is putting a developer back in control.
+        </p>
+        <p>
+          Build instructions are written in Nix language, which allows you to
+          extend or change existing build instruction in any way you wish. This
+          makes Nix a powerful build tool that can be used in both simple and
+          complex projects.
+        </p>
+      </div>
+    </div>
+  </div>
+  <div>
+    [% PROCESS svg path="site-styles/assets/gfx-overview-cloud.svg" %]
+    <div>
+      <h2>
+        From zero to cloud.
+      </h2>
+      <ul>
+        <li>cloud images</li>
+        <li>docker images</li>
+        <li>devops</li>
+      </ul>
+      <p>
+        Use the same tool to build one project or even one hundred projects,
+        configure machines and deploy to the cloud. Thanks to the extensible
+        Nix language, it's easy to compose parts together with little overhead.
+      </p>
+      <div>
+        <p>
+          Use one tool to develop, test and build Amazon EC2, Google Cloud,
+          Azure, Virtualbox, KVM and other images in a declarative way is a
+          DevOps dream.
+        </p>
+        <p>
+          Forget <code>Dockerfile</code> and
+          have a peace of mind with declarative approach Nix gives you. Nix
+          automatically calculates the optimal layer layout and only includes
+          software needed.
+          <strong>Nix and Docker is a match made in heaven.</strong>
+        </p>
+      </div>
+    </div>
+  </div>
+
+  <div>
+    [% PROCESS svg path="site-styles/assets/gfx-overview-gears.svg" %]
+    <div>
+      <h2>
+        Is the previous version better? Rollback anytime.
+      </h2>
+      <p>
+        Packages are never overwritten, and old versions are always stored.
+        Rollback to any previous version within milliseconds.
+      </p>
+      <ul>
+        <li>generations</li>
+        <li>garbage collection</li>
+        <li>atomic</li>
+        <li>nix store</li>
+      </ul>
+      <div>
+        <p>
+          Since Nix never overwrites files in the Nix store, but just adds new
+          versions in different paths, the upgrades are atomic. During an
+          upgrade, there is no time window in which some files exists from the
+          old version and some files from the new version - which would be bad
+          because a program might crash if it's restarted during that period.
+        </p>
+        <p>
+          And since nothing is overwritten, the old versions are still there
+          after an upgrade. This means that you can rollback to the old
+          version. Rollbacks are very fast: it doesn't involve lots of files
+          having to be restored from copies.
+        </p>
+      </div>
+    </div>
+  </div>
+
+  <div>
+    <!-- hacking, fun, party -->
+    [% PROCESS svg path="site-styles/assets/gfx-overview-gears.svg" %]
+    <div>
+      <h2>
+        It is important to experiment.
+      </h2>
+      <ul>
+        <li>generations</li>
```

Comment:
We don't need to mention "generations", they're just an implementation aspect of the upgrade/rollback mechanism.